syntax = "proto3";

package sr_output_proto;

message Pose {
  //position info
  double p_x = 1;
  double p_y = 2;
  double p_z = 3;
  //orientation info
  double o_x = 4;
  double o_y = 5;
  double o_z = 6;
  double o_w = 7;
}

message ImgData{
    bytes img_data = 1;
    int32 rows = 2;
    int32 columns = 3;
}
//uncompressed mesh
message MeshData{
    repeated double v_x = 1;
    repeated double v_y = 2;
    repeated double v_f = 3;
    repeated int32  c_r = 4;
    repeated int32  c_g = 5;
    repeated int32  c_b = 6;
    repeated int64  f_1 = 7;
    repeated int64  f_2 = 8;
    repeated int64  f_3 = 9;
}

//for sending VB lists
message VB {
    int32 x = 1;
    int32 y = 2;
    int32 z = 3;
}

//compressed mesh
//1 active, 2 inactive, 3 vb_list
message CompressMeshData{
    bytes draco_data =1;
    uint32 active = 2;
    uint32 request_id=3;
    uint32 chunk_id=4;
    uint32 max_chunk=5;
    repeated VB vbs=6;
}

